src/groovy/org/pih/warehouse/api/PutawayItem.groovy

```diff
@@ -114,12 +114,12 @@ class PutawayItem {
                 "recipient.id"                : recipient?.id,
                 "recipient.name"              : recipient?.name,
                 currentBins                   : currentBins,
-                preferredBin                  : preferredBin,
+                preferredBin                  : preferredBin?.name,
                 currentBinsAbbreviated        : currentBinsAbbreviated,
                 "putawayFacility.id"          : putawayFacility?.id,
                 "putawayFacility.name"        : putawayFacility?.name,
-                "putawayLocation.id"          : putawayLocation?.id,
-                "putawayLocation.name"        : putawayLocation?.name,
+                "putawayLocation.id"          : putawayLocation?.id ?: preferredBin?.id,
```

Comment:
I don't like this. This indicates that despite `pytawatLocation` is null, the returned map will state that it is existing. I think the `preferredBin.id` should be a separate field in this map.